app/controllers/hiring_staff/vacancies/important_dates_controller.rb

```diff
@@ -0,0 +1,52 @@
+class HiringStaff::Vacancies::ImportantDatesController < HiringStaff::Vacancies::ApplicationController
+  before_action :redirect_unless_vacancy
+  before_action :set_up_important_dates_form, only: %i[update]
+  before_action only: %i[update] do
+    save_vacancy_as_draft_if_save_and_return_later(@important_dates_form.params_to_save, @vacancy)
+  end
+
+  def show
+    @important_dates_form = ImportantDatesForm.new(@vacancy.attributes.symbolize_keys)
+  end
+
+  def update
+    @important_dates_form.id = @vacancy.id
+    @important_dates_form.status = @vacancy.status
+
+    if @important_dates_form.complete_and_valid?
+      update_vacancy(@important_dates_form.params_to_save, @vacancy)
+      update_google_index(@vacancy) if @vacancy.listed?
+      return redirect_to_next_step_if_save_and_continue(@vacancy.id)
+    end
+
+    render :show
+  end
+
+  private
+
+  def set_up_important_dates_form
```

Comment:
Why don't you put all this logic in the form object? Controllers should be skinnier. This could be a refactoring for later